components/tools/OmeroJava/test/integration/gateway/GatewayUsageTest.java

```diff
@@ -73,6 +74,31 @@ public void testLoginWithArgs()
     }
     
     @Test
+    public void testLoginWithSessionID() throws DSOutOfServiceException {
+        omero.client client = new omero.client();
+        String[] args = new String[4];
+        args[0] = "--omero.host=" + client.getProperty("omero.host");
+        args[1] = "--omero.port=" + client.getProperty("omero.port");
+        args[2] = "--omero.user=root";
+        args[3] = "--omero.pass=" + client.getProperty("omero.rootpass");
+        LoginCredentials c = new LoginCredentials(args);
+        Gateway gw = new Gateway(new SimpleLogger());
+        ExperimenterData root = gw.connect(c);
+        String sessionId = gw.getSessionId(root);
+
+        Gateway gw2 = new Gateway(new SimpleLogger());
+        JoinSessionCredentials c2 = new JoinSessionCredentials(sessionId,
+                client.getProperty("omero.host"), Integer.parseInt(client
+                        .getProperty("omero.port")));
+        ExperimenterData root2 = gw2.connect(c2);
+        Assert.assertNotNull(root2);
+        Assert.assertEquals(gw2.getSessionId(root2), sessionId);
+        gw2.disconnect();
```

Comment:
If something throws re: `gw2`, `gw1` will not be closed which could hang the test.

Reply:
Could make the Gateway AutoCloseable ( c31e719 ), would make things easier. Any objections?

Reply:
No objections. :smiley: